### DISCUSSION TITLE
Libre2 connection issues after working fine for days

### DISCUSSION THREAD

**Discussion by dagord** [NightscoutFoundation/xDrip#2891] [Q&A]
Hello.
My girlfriend uses Libre2 and this was our very first attempt to use direct connection with xDrip.
In order to simply things, we started the sensor with FSL1 standalone reader, then after a few hours we started it on xDrip following this [walkthrough](https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip) (we have not installed the FSL app at all): it got conneced at the first attempt and worked pretty well until yesterday evening (more than 10 days), when it suddenly stopped sending values to xDrip. NFC scan would work normally both on FSL1 reader and xDrip, so the problem was the bluetooth connection itself.
I tried:
- cycling BT connection
- cycling airplane mode
- restarting the phone
- deleting xDrip cache
- evntually stopping the sensor (on xDrip) and starting it again

I finally gave it up, but after a couple of hours the sensor was back on by itself, and has been working normally all along the night. Currently my girlfriend is at work but she texted me that it is still behaving normally.
So what's happened? There's a way to troubleshoot this behaviour?

I'll try to provide logs (if useful) when she comes back home.

**Comment by Navid200** [comment]
@psonnera

**Comment by psonnera** [comment]
Check your bluetooth settings? (scroll down) https://xdrip.readthedocs.io/en/latest/install/libre2/#setup-xdrip

**Reply by dagord** [reply]
Hello.
my BT settings are slightly different: "trust auto connect" and "use background scan" are flagged as suggested on [minimallooper](https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip#viewer-4l2fb) website. By the way, after being reconnected, without any change in the settings, the sensor happily reached its expiration date.

We had another issue with the new sensor, which was detected as Libre1, but a quick search lead me to [this post](https://github.com/NightscoutFoundation/xDrip/discussions/2845#discussioncomment-5995246), so I upgraded xDrip and OOP2; now it's wotking fine again
